pkgs/compiler-lib/compiler/commands/test.rkt

```diff
@@ -274,9 +281,12 @@
                       (format "~s" d)
                       (format "~s" make?)
                       (format "~s" (and load-errortrace? errortrace-module-path))
+                      (format "~s" (cond
+                                     [(current-test-invocation-directory) => path->string]
```

Comment:
This should be `path->bytes` with `bytes->path` on the other end, `path->string` does not always round-trip with `string->path`.

Reply:
Indeed. I notice that `normalize-module-path` uses `path->string` and is also put in the subprocess arguments. Something to audit/fix another day?

`git range-diff`:
```
1:  b3a32b284c ! 1:  50ed11c902 raco test: set more current-test-invocation-directory
    @@ Commit message
     
         This adds the effects of df01b0d1dc (raco test: set
         current-test-invocation-directory, 2024-05-07) to --place and --process
    -    mode, too; as a result, parallel invocations with -j work, too.
    +    mode, too; as a result, parallel invocations with -j work, too. To pass
    +    the path to a subprocess, we need to convert to bytes (since paths do
    +    not round trip through strings).
     
         The changes to the subprocess argv processing and place channel received
         list processing do perhaps beg the question: is the protocol for
    @@ pkgs/compiler-lib/compiler/commands/test.rkt: (define rt-module (read (open-inpu
        (define make? (read (open-input-string (vector-ref argv 4))))
        (define errortrace-path-or-false (read (open-input-string (vector-ref argv 5))))
     -  (define args (list-tail (vector->list argv) 6))
    -+  (define test-invocation-path (read (open-input-string (vector-ref argv 6))))
    ++  (define test-invocation-path (bytes->path (read (open-input-string (vector-ref argv 6)))))
     +  (define args (list-tail (vector->list argv) 7))
     +
     +  (current-test-invocation-directory test-invocation-path)
    @@ pkgs/compiler-lib/compiler/commands/test.rkt: (define ps
                            (format "~s" make?)
                            (format "~s" (and load-errortrace? errortrace-module-path))
     +                      (format "~s" (cond
    -+                                     [(current-test-invocation-directory) => path->string]
    ++                                     [(current-test-invocation-directory) => path->bytes]
     +                                     [else #f]))
                            args)))
                 (define proc (list-ref ps 4))
```

Reply:
> Indeed. I notice that `normalize-module-path` uses `path->string` and is also put in the subprocess arguments. Something to audit/fix another day?

Yes, agreed. Thanks for the fixing the directory, at least!